okapi-core/src/main/java/org/folio/okapi/MainVerticle.java

```diff
@@ -253,78 +250,71 @@ public void start(Promise<Void> promise) {
-    tenantManager.get(XOkapiHeaders.SUPERTENANT_ID).onComplete(gres -> {
-      if (gres.succeeded()) { // we already have one, go on
-        logger.info("checkSuperTenant: Already have " + XOkapiHeaders.SUPERTENANT_ID);
-        Tenant st = gres.result();
-        Set<String> enabledMods = st.getEnabled().keySet();
-        if (enabledMods.contains(okapiModule)) {
-          logger.info("checkSuperTenant: enabled version is {}", okapiModule);
-          promise.complete();
-          return;
-        }
-        // Check version compatibility
-        String enver = "";
-        for (String emod : enabledMods) {
-          if (emod.startsWith("okapi-")) {
-            enver = emod;
-          }
-        }
-        final String ev = enver;
-        logger.debug("checkSuperTenant: Enabled version is '{}', not '{}'",
-            ev, okapiModule);
-        // See Okapi-359 about version checks across the cluster
-        if (ModuleId.compare(ev, okapiModule) >= 4) {
-          logger.warn("checkSuperTenant: This Okapi is too old,"
-                  + "{} we already have {} in the database. Use that!",
-              okapiVersion, ev);
-          promise.complete();
-          return;
+    tenantManager.get(XOkapiHeaders.SUPERTENANT_ID).onSuccess(tenant -> {
+      // we already have one, go on
+      logger.info("checkSuperTenant: Already have " + XOkapiHeaders.SUPERTENANT_ID);
+      Set<String> enabledMods = tenant.getEnabled().keySet();
+      if (enabledMods.contains(okapiModule)) {
+        logger.info("checkSuperTenant: enabled version is {}", okapiModule);
+        promise.complete();
+        return;
+      }
+      // Check version compatibility
+      String enver = "";
+      for (String emod : enabledMods) {
+        if (emod.startsWith("okapi-")) {
+          enver = emod;
         }
-        logger.info("checkSuperTenant: Need to upgrade the stored version from {} to {}",
-            ev, okapiModule);
-        // Use the commit, easier interface.
-        // the internal module can not have dependencies
-        // See Okapi-359 about version checks across the cluster
-        tenantManager.updateModuleCommit(st, ev, okapiModule).onComplete(ures -> {
-          if (ures.failed()) {
-            promise.fail(ures.cause());
-            return;
-          }
-          logger.info("Upgraded the InternalModule version from '{}' to '{}' for {}",
-              ev, okapiModule, XOkapiHeaders.SUPERTENANT_ID);
-          promise.complete();
-        });
+      }
+      final String ev = enver;
+      logger.debug("checkSuperTenant: Enabled version is '{}', not '{}'",
+          ev, okapiModule);
+      // See Okapi-359 about version checks across the cluster
+      if (ModuleId.compare(ev, okapiModule) >= 4) {
+        logger.warn("checkSuperTenant: This Okapi is too old,"
+                + "{} we already have {} in the database. Use that!",
+            okapiVersion, ev);
+        promise.complete();
```

Comment:
This PR did not change this part, but I am wondering if we should consider downgrade Okapi version for superteant so it matches the actual running version?

Reply:
We could it it's just the Okapi module .. But we do have to be careful.. Eg ModuleDescriptors with new properties that suddenly will not be recognized by Okapi anymore.. I'll see what I can do as part of the "upgrade Okapi module for all tenants".. https://issues.folio.org/browse/OKAPI-930